generator/src/main/java/org/apache/kafka/message/checker/CheckerUtils.java

```diff
@@ -125,10 +126,17 @@ static MessageSpec readMessageSpecFromString(String contents) {
      * @param filePath   The file to read from remote git repo.
      * @return                     The file contents.
      */
-    static String GetDataFromGit(String filePath, Path gitPath) throws IOException {
+    static String getDataFromGit(String filePath, Path gitPath) throws IOException {
         Git git = Git.open(new File(gitPath + "/.git"));
         Repository repository = git.getRepository();
-        Ref head = git.getRepository().getRefDatabase().firstExactRef("refs/heads/trunk");
```

Comment:
Maybe we can address this by introducing a new argument, ref, to the tool? This argument would allow specifying the exact ref you want to trace. By default, it could be set to refs/heads/trunk, but during testing, we could pass HEAD instead.